src/monster.cpp

```diff
@@ -702,6 +702,11 @@ monster_attitude monster::attitude(player *u) const
                 effective_morale -= 5;
             }
         }
+        if (type->in_species("FUNGUS" && u->has_trait("MYCUS_THRESH"))) 
```

Comment:
Quite interesting that this compiles. My mysterious glass ball tells me it's definitely not what you want.

The condition is `type->in_species("FUNGUS" && u->has_trait("MYCUS_THRESH"))`, which is a call to `mtype::in_species` with `"FUNGUS" && u->has_trait("MYCUS_THRESH")` as argument. That argument is a `boolean and` applied to a string literal (`"FUNGUS"`) and a boolean returned by `Creature::has_trait`. The string literal (actually an array) decays to a pointer, which is converted to a boolean and is always `true` (because a string literal is per definition never a pointer to null). The result of the `boolean and` is a boolean.

But what `in_species` function is called with a boolean? Turns out there is a `mtype::in_species(int)` function. This one is called, with the boolean converted to either 0 or 1.

The mentioned mysterious glass ball says you wanted to write `type->in_species("FUNGUS") && u->has_trait("MYCUS_THRESH")`.